lib/widget/show_appointment.dart

```diff
@@ -0,0 +1,142 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:flutter/material.dart';
+import '../services/interfaces/calculation_service_interface.dart';
+import '../services/interfaces/firebase_service_interface.dart';
+import '../services/service_locator.dart';
+
+class ShowAppointment extends StatefulWidget {
+  @override
+  State<StatefulWidget> createState() => _ShowAppointmentState();
+}
+
+class _ShowAppointmentState extends State<ShowAppointment> {
+  final IFirebaseService _firebaseService = locator<IFirebaseService>();
+  final ICalculationService _calculationService =
+      locator<ICalculationService>();
+  List<Map<String, dynamic>> appointments = <Map<String, dynamic>>[];
+
+  __initData() {
+    var data = _getAppointments();
+    print('IS NULL  ? ? ?$appointments');
+    data.then((value) => appointments.addAll(value));
+    print('aw my god$appointments');
+  }
+
+  Future<List<Map<String, dynamic>>> _getAppointments() async {
+    final _appointments = await _firebaseService.getAppointments();
+    print('=[]= showAppointment = $_appointments');
+    print('${_appointments.length}');
+    return _appointments;
+  }
+
+  var _userId;
+  var _userCollection;
+  @override
+  void initState() {
+    super.initState();
+    __initData();
+    var userId = _firebaseService.getUserId();
+    setState(() {
+      _userId = userId;
+      _userCollection = _firebaseService.getCollectionSnapshotByDocId(
+          collection: 'Users', docId: _userId);
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return StreamBuilder<DocumentSnapshot>(
+        stream: _userCollection,
+        builder: (context, userCollection) {
+          if (!userCollection.hasData) {
+            return Center(
+              child: Column(
+                children: <Widget>[
+                  CircularProgressIndicator(),
+                  Text('Loading...'),
+                ],
+              ),
+            );
+          } else {
+            return FutureBuilder<Map<String, dynamic>>(
+                future:
+                    _firebaseService.getLatestAnSubCollection(userId: _userId),
+                builder: (context, anSubCollection) {
+                  if (!anSubCollection.hasData) {
+                    return Center(
+                      child: Column(
+                        children: <Widget>[
+                          CircularProgressIndicator(),
+                          Text('Loading...'),
+                        ],
+                      ),
+                    );
+                  }
```

Comment:
Why do we need these 2 builders

Reply:
The builder will load on that page, equivalent to hot reload.

Reply:
But we don't need other data except the data from `Appointments` Collection.

Reply:
In line 63, it's gonna consume the query resource

Reply:
If there is no builder for loading, it will need to be hot reload again.

Reply:
I'll try to fix it again.